numpy/_array_api/_statistical_functions.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from ._array_object import ndarray
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ._types import Optional, Tuple, Union, array
+
+import numpy as np
+
+def max(x: array, /, *, axis: Optional[Union[int, Tuple[int, ...]]] = None, keepdims: bool = False) -> array:
```

Comment:
```suggestion
from numpy.typing import _Shape  # i.e. `Tuple[int, ...]

def max(x: array, /, *, axis: None | int | _Shape = None, keepdims: bool = False) -> array:
```
There is a `_Shape` type-alias in `numpy.typing` that might be of use here.
Helps a bit with keeping the verbosity down.

Reply:
So far I've kept the type hints in the array API namespace self-contained. I think it's useful to help avoid NumPy types from slipping in, and it also could make it easier for others looking to copy this implementation for their own array API compatible submodule (I know the CuPy team is planning on doing this once this lands). I also think it's useful to have the type signatures match those in the spec as closely as possible.